### ISSUE TITLE
[bug]  no-unused-props: False positive when props are used via destructuring in downstream code

### ISSUE THREAD

**Issue by marcospgp** [Rel1cx/eslint-react#1268]
### Describe the bug

Describe the bug

  The @eslint-react/no-unused-props rule incorrectly flags props as unused when they are defined in a
   type that is later destructured or spread in downstream code.

  To Reproduce

```
  type BaseOptions = {
    dismissible?: boolean;
    fitContent?: boolean;
  };

  type CustomOptions = {
    onClose?: () => void;
  } & BaseOptions;

  // ESLint flags dismissible and fitContent as unused here
  function ComponentA(props: BaseOptions) {
    return <div>{/* ... */}</div>;
  }

  // ESLint flags onClose as unused here
  function ComponentB(props: CustomOptions) {
    return <div>{/* ... */}</div>;
  }

  // But the props ARE actually used via destructuring
  function createThing(options: CustomOptions) {
    const { onClose, dismissible, fitContent, ...rest } = options;

    // dismissible used here
    if (dismissible) {
      // ...
    }

    // fitContent used here
    if (fitContent) {
      // ...
    }

    // onClose used here
    onClose?.();
  }
```

  Expected behavior

  Props should not be flagged as unused when they are used in downstream code via destructuring or
  spreading, even if not directly used in the component where the type is first referenced.

  Environment
  - @eslint-react/eslint-plugin: ^2.0.6
  - ESLint: latest
  - TypeScript: latest

  Additional context

  This appears in real-world code where:
  1. Base types define common props
  2. Components accept these types but don't directly use all props
  3. Props are extracted and used in imperative APIs or factory functions

  The rule seems to only analyze direct usage within the component function rather than tracking how
  the type's props flow through the codebase.

### Platform and versions

```Text
see above
```

### Stack trace

```Text

```
